crates/nu-plugin/src/plugin/mod.rs

```diff
@@ -84,12 +92,45 @@ pub trait PluginEncoder: Encoder<PluginInput> + Encoder<PluginOutput> {
     fn name(&self) -> &str;
 }
 
-fn create_command(path: &Path, shell: Option<&Path>) -> CommandSys {
-    log::trace!("Starting plugin: {path:?}, shell = {shell:?}");
+#[cfg(unix)]
+fn quote_arg(arg: impl AsRef<OsStr>) -> OsString {
+    use std::os::unix::ffi::{OsStrExt, OsStringExt};
+    let arg_bytes = arg.as_ref().as_bytes();
+    let mut out: Vec<u8> = Vec::with_capacity(arg_bytes.len() + 10 /* just a guess */);
+    out.push(b'"');
+    for &byte in arg_bytes {
+        if byte == b'"' {
+            // Add backslash escape
+            out.push(b'\\');
+        }
+        out.push(byte);
+    }
+    out.push(b'"');
+    OsString::from_vec(out)
+}
 
-    // There is only one mode supported at the moment, but the idea is that future
-    // communication methods could be supported if desirable
-    let mut input_arg = Some("--stdio");
+#[cfg(windows)]
+fn quote_arg(arg: impl AsRef<OsStr>) -> OsString {
+    use std::os::windows::ffi::{OsStrExt, OsStringExt};
+    let arg = arg.as_ref();
+    // Windows uses wide characters (ideally UTF-16) for native strings
+    let mut out: Vec<u16> = Vec::with_capacity(arg.len() + 10 /* just a guess */);
+    out.push(b'"' as u16);
+    for wide_ch in arg.encode_wide() {
+        if wide_ch == b'"' as u16 {
+            // Add backslash escape
+            out.push(b'\\' as u16);
+        }
+        out.push(wide_ch);
+    }
+    out.push(b'"' as u16);
+    OsString::from_wide(&out)
+}
```

Comment:
Having to escape things is unfortunate, but I think this is correct..

Reply:
oh, I meant to revisit this, because I think the escaping is not proper for `cmd`. The BatBadBut issue reminded me

Reply:
That said, maybe we don't actually need to use the shell? Unix shell script execution is handled by the OS as long as it's +x, and on Windows it looks like Rust does its own stuff to make sure batch files can execute.

What do you think?

Reply:
Ok, I just decided to try it. I also added a few other filetypes that might be used :shrug: I think we'll have to maintain that list manually, but it should work well enough

Reply:
> That said, maybe we don't actually need to use the shell?

If it looks like it works, then yeah that sounds good.

Reply:
Yes, it seems to work, the python plugin works on windows, including from a batch file